core/src/test/kotlin/generator/TestRunner.kt

```diff
@@ -0,0 +1,89 @@
+package generator
+
+import org.jetbrains.dokka.*
+import org.jetbrains.dokka.utilities.DokkaConsoleLogger
+import org.jetbrains.dokka.utilities.DokkaLogger
+import java.io.File
+import java.nio.charset.Charset
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+
+
+object TestRunner {
+    private val testDir: Path? by lazy {
+        Files.createTempDirectory("tests").also { it.toFile().deleteOnExit() }
+    }
+
+    fun generateTestDokka(
+        name: String,
+        query: String,
+        checks: Map<DokkaTestGenerator.Companion.Stage, (DokkaTestGenerator.State) -> Unit?>,
+        targets: List<String> = listOf("jvm"),
+        platforms: List<String> = listOf("jvm"),
+        logger: DokkaLogger = DokkaConsoleLogger
+    ) {
+        val root = testDir!!.resolve(name)
+        root.deleteContents()
+
+        val output = root.resolve("out")
+        Files.createDirectories(output)
+
+        platforms.map { it.toPlatform() }
+            .forEach {
+                output.deleteContents()
+
+                val config = DokkaConfigurationImpl(
+                    outputDir = output.toString(),
+                    format = "html",
+                    generateIndexPages = true,
+                    passesConfigurations = listOf(
+                        PassConfigurationImpl(
+                            moduleName = name,
+                            sourceRoots = listOf(root).map { SourceRootImpl(it.toString()) },
+                            analysisPlatform = it,
+                            targets = targets
+                        )
+                    )
+                )
+
+                query.toFileMap().materializeFiles(root)
+
+                DokkaTestGenerator(config, logger, checks).generate()
+            }
+    }
+
+    fun String.toFileMap(): Map<String, String> = if (this.trim().isNotEmpty()) {
+        this.replace("\r\n", "\n")
+            .split("\n/")
+            .map {
+                it.split("\n".toRegex(), 2)
+                    .let {
+                        it.first().trim().removePrefix("/") to it.last().trim()
+                    }
+            }
+            .toMap()
+    } else {
+        mapOf()
+    }
```

Comment:
`split("\n", limit = 2).let { mapOf(it.first() to it.last().trim()) }`

Reply:
Removed.